src/machine.rs

```diff
@@ -251,6 +251,11 @@ pub struct Evaluator<'mir, 'tcx> {
     /// The "time anchor" for this machine's monotone clock (for `Instant` simulation).
     pub(crate) time_anchor: Instant,
 
+    /// The approximate system time when "time anchor" was created. This is used
+    /// for converting system time to monotone time so that we can simplify the
+    /// thread scheduler to deal only with a single representation of time.
```

Comment:
Something seems extremely fishy if we ever have to convert monotone time to absolute time... that's just fundamentally against the principle of monotone time.

Reply:
The reason why I am doing the conversion is to use only one notion of time in `src/sync.rs`. If you think that this is not a good idea, I can change it (I do not have a strong opinion here).

Reply:
I'll need to take a closer look at what you are doing there, didn't get there yet.